the-collab-lab/tcl-66-smart-shopping-list
# Sc issue 22

## Description

Create Toast.jsx, ConfirmToast.jsx, useToast hook, Sprite.jsx and sprite.svg. Implementing these new components, hooks and the reusable icon file,  stylish and design consistent messages replace window.alert and window.confirm methods from the browser. The user either sees Toast, which gives them a message and fades away,  or the ConfirmationToast which allows them to proceed or cancel based on a warning message. The messages are more conveniently placed in the view as well as matching the feel of the app.

Each component has color, icon and message versatility so that they can be reused in different scenarios; for example an error, a warning, or a success. The Confirmation toast also takes two call backs, so that different actions can me taken depending on if the user confirms or cancels the action. The regular toast also has a dismissible feature,  the user can close it out instead of waiting for its functionality to remove it.

I was not able to implement the animation suggested in the design (slide in and back out). That will be an issue for the future  when I have many more hours of tailwind under my belt 😅.

## Related Issue

closes #49 

## Acceptance Criteria

## Acceptance criteria
- [x] Use FlowBite and Figma design
- [x] Create a Toast component
- [x] Close Button/ Dismissible
- [x] Message
- [x] Animate as per Figma Design
- [x] Create ConfirmToast Component
- [x] Message
- [x] Cancel/ Confirm

## Type of Changes

`enhancement`

## Updates

### Before

<img width="833" alt="Screenshot 2024-04-02 at 10 17 17 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/108297341/b3f56aba-4c80-4b9a-8c85-8bc058c48718">

<img width="627" alt="Screenshot 2024-04-02 at 10 17 25 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/108297341/df4a9863-c278-48d1-a6bd-51b6e52935ce">

### After

<img width="605" alt="Screenshot 2024-04-02 at 10 16 56 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/108297341/2c0cbd49-afbc-40e3-95e0-4b47d8feb025">

<img width="416" alt="Screenshot 2024-04-02 at 10 16 42 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/108297341/b5d6bd51-1e10-4c89-8ff3-9a5da21adbfe">

## Testing Steps/QA material
* git pull from main
* git checkout sc-issue-24
* npm start
* the existing alerts and confirmations were on deleting lists, sharing lists, adding items, and deleting items -- check out each of these scenarios, including leaving fields blank, creating a duplicate item, deleting a shared list and a non shared list, inviting a friend without completing the fields...


###NOTE
There are two places where the Toast will not work because of how our logic/ app is set up. 
* When adding the first item to a list -- I couldn't quite see what was happening to prevent the toast.
* When sharing a list, the page reloads so again the Toast is not visible. I implemented them but commented out the alerts, I was not sure what everyone would think here and wanted to wait for the team before making big decisions.
